modules/guides/pages/bulk-operations.adoc

```diff
@@ -0,0 +1,420 @@
+----
+
+{github}
+--
+
+Python::
++ 
+--
+. Create a dictionary of structured JSON documents.
+
+. Pass the dictionary to the `insert_multi()` function. This will insert all the documents in the database.
+
+. A dictionary of `MutationResult` objects is returned.
+
+CAUTION: `CBCollection.insert_multi` is a xref:python-sdk:project-docs:compatibility.adoc#interface-stability[volatile] API call that is still in flux and may likely be changed.
+
+{kv-bulk-insert-example}
+
+[source,python]
+----
+include::python-sdk:hello-world:example$kv_bulk_hello_world.py[tags=kv-users;kv-bulk-insert,indent=0]
+----
+
+{github}
+--
+====
+
+== Reading Multiple Documents
+
+To read multiple documents in Couchbase perform a bulk `get` operation.
+
+[{tabs}]
+====
+cbc::
++ 
+--
+Use the `cbc cat` command to retrieve multiple documents by their IDs. 
+
+{kv-bulk-get-example}
+
+[source,shell]
+----
+include::example$kv/kv-cbc.sh[tag=cbc-bulk-get]
+----
+
+.Result
+[source,console]
+----
+0                    CAS=0x16be2392ca2e0000, Flags=0x0, Size=904, Datatype=0x01(JSON)
+{
+  "name": "Keon Hoppe",
+  "addresses": [
+    {
+      "type": "home",
+      "address": "222 Sauer Neck",
+      "city": "London",
+      "country": "United Kingdom"
+    },
+  ],
+  ...
+}
+1                    CAS=0x16be2392c9870000, Flags=0x0, Size=697, Datatype=0x01(JSON)
+{
+  "name": "Rigoberto Bernier",
+  "addresses": [
+    {
+      "type": "home",
+      "address": "0622 Adams Mills",
+      "city": "Manchester",
+      "country": "United Kingdom"
+    }
+  ],
+  ...
+}
+----
+--
+
+.NET::
++ 
+--
+. Initialize a list of `IGetResult` tasks.
+
+. Perform a `get` operation on each document and store the results in the tasks list.
+
+. Wait for all the tasks to complete before accessing the results.
+
+{kv-bulk-get-example}
+
+[source,csharp]
+----
+include::dotnet-sdk:hello-world:example$KvBulkHelloWorld.csx[tag=kv-users,indent=0]
+
+include::dotnet-sdk:hello-world:example$KvBulkHelloWorld.csx[tag=kv-bulk-get,indent=0]
+
+----
+
+{github}
+--
+
+Java::
++ 
+--
+Using the `reactor.core.publisher.Flux` reactive library, call the `fromIterable()` method to perform multiple `get` operations.
+
+{kv-bulk-get-example}
+
+[source,java]
+----
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-users,indent=0]
+
+include::java-sdk:hello-world:example$KvBulkHelloWorld.java[tag=kv-bulk-get,indent=0]
+
+----
+
+{github}
+--
+
+Node.js::
++ 
+--
+Perform a `get` operation on each document and wait for all the promises to complete before accessing the results.
+
+{kv-bulk-get-example}
+
+[source,nodejs]
+----
+include::nodejs-sdk:hello-world:example$kv-bulk-hello-world.js[tags=kv-users;kv-bulk-get,indent=0]
+----
+
+{github}
+--
+
+Python::
++ 
+--
+. Pass some document IDs to fetch to the `get_multi()` function. This will fetch the documents from the database.
+
+. A dictionary of `GetResult` objects is returned.
```

Comment:
Remove number from second paragraph because it's a result, not a "do".
Consequently, remove number from first paragraph, because it then becomes the only step.

```suggestion
Pass some document IDs to fetch to the `get_multi()` function. This will fetch the documents from the database.

A dictionary of `GetResult` objects is returned.
```